src/ja/readmangaat/src/eu/kanade/tachiyomi/extension/ja/readmangaat/ReadMangaAt.kt

```diff
@@ -0,0 +1,324 @@
+            setUrlWithoutDomain(attr("abs:href"))
+            title = text()
+        }
+    }
+
+    override fun popularMangaNextPageSelector(): String? = null
+
+    // =============================== Latest ===============================
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return if (page > 1) {
+            val apiBody = FormBody.Builder().apply {
+                add("action", "z_do_ajax")
+                add("_action", "loadmore")
+                add("p", page.toString())
+            }.build()
+
+            POST(ajaxUrl, apiHeaders, apiBody)
+        } else {
+            GET(baseUrl, headers)
+        }
+    }
+
+    override fun latestUpdatesSelector(): String = "div.row > div.col-sm-6"
+
+    override fun latestUpdatesFromElement(element: Element): SManga = SManga.create().apply {
+        with(element.selectFirst("a:has(img)")!!) {
+            thumbnail_url = selectFirst("img")!!.attr("abs:src")
+            setUrlWithoutDomain(attr("abs:href"))
+            title = attr("title")
+        }
+    }
+
+    override fun latestUpdatesNextPageSelector(): String = ".text-center > a:contains(Load More)"
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        if (response.request.method == "GET") {
+            return super.latestUpdatesParse(response)
+        }
+
+        val data = response.parseAs<LatestDto>()
+        val mangaList = Jsoup.parseBodyFragment(data.mes)
+            .select("div.col-sm-6")
+            .map(::latestUpdatesFromElement)
+        val hasNextPage = data.going == "yes"
+
+        return MangasPage(mangaList, hasNextPage)
+    }
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = baseUrl.toHttpUrl().newBuilder().apply {
+            val genreFilter = filters.filterIsInstance<GenreFilter>().first()
+
+            if (genreFilter.toUriPart().isNotBlank()) {
+                addPathSegment("genres")
+                addEncodedPathSegment(genreFilter.toUriPart())
+            } else {
+                addQueryParameter("s", query)
+            }
+
+            addPathSegment("")
+            if (page > 1) {
+                addPathSegment("page")
+                addPathSegment(page.toString())
+                addPathSegment("")
+            }
+        }.build()
+
+        return GET(url, headers)
+    }
+
+    override fun searchMangaSelector(): String = ".row > .col-6 > .entry-ma"
+
+    override fun searchMangaFromElement(element: Element): SManga =
+        latestUpdatesFromElement(element)
+
+    override fun searchMangaNextPageSelector(): String = ".pagination > span.current + a"
+
+    // =============================== Filters ==============================
+
+    override fun getFilterList(): FilterList = FilterList(
+        Filter.Header("Ignored when using text search"),
```

Comment:
The code in `searchMangaRequest` prioritizes the genre filter over text search though?